applications/main/infrared/scenes/infrared_scene_universal_more_devices.c

```diff
@@ -0,0 +1,178 @@
+#include "../infrared_app_i.h"
+#include <storage/storage.h>
+#include <furi.h>
+#include <dolphin/dolphin.h>
+
+
+static void
+    infrared_scene_universal_more_devices_callback(void* context, int32_t index, InputType type) {
+    UNUSED(type);
+    InfraredApp* infrared = context;
+    uint32_t event = infrared_custom_event_pack(InfraredCustomEventTypeButtonSelected, index);
+    view_dispatcher_send_custom_event(infrared->view_dispatcher, event);
+}
+
+static void infrared_scene_universal_more_devices_progress_back_callback(void* context) {
+    InfraredApp* infrared = context;
+    uint32_t event = infrared_custom_event_pack(InfraredCustomEventTypeBackPressed, -1);
+    view_dispatcher_send_custom_event(infrared->view_dispatcher, event);
+}
+
+static void
+    infrared_scene_universal_more_devices_show_popup(InfraredApp* infrared, uint32_t record_count) {
+    ViewStack* view_stack = infrared->view_stack;
+    InfraredProgressView* progress = infrared->progress;
+    infrared_progress_view_set_progress_total(progress, record_count);
+    infrared_progress_view_set_back_callback(
+        progress, infrared_scene_universal_more_devices_progress_back_callback, infrared);
+    view_stack_add_view(view_stack, infrared_progress_view_get_view(progress));
+    infrared_play_notification_message(infrared, InfraredNotificationMessageBlinkStartSend);
+}
+
+static void infrared_scene_universal_more_devices_hide_popup(InfraredApp* infrared) {
+    ViewStack* view_stack = infrared->view_stack;
+    InfraredProgressView* progress = infrared->progress;
+    view_stack_remove_view(view_stack, infrared_progress_view_get_view(progress));
+    infrared_play_notification_message(infrared, InfraredNotificationMessageBlinkStop);
+}
+
+static int32_t infrared_scene_universal_more_devices_task_callback(void* context) {
+    InfraredApp* infrared = context;
+    const InfraredErrorCode error = infrared_brute_force_calculate_messages(infrared->brute_force);
+    view_dispatcher_send_custom_event(
+        infrared->view_dispatcher,
+        infrared_custom_event_pack(InfraredCustomEventTypeTaskFinished, 0));
+
+    return error;
+}
+
+void infrared_scene_universal_more_devices_on_enter(void* context) {
```

Comment:
looking at how the blocking task works i think this can be done without loading the db twice, ill play around with it a bit

Reply:
hopefully, instead of loading it in on_enter, we can do the "initial" load in the blocking task callback and add the buttons from there. as for the first button detected not counting to total count, can just edit the logic in calculate_messages() since this is extra functionality added here anyway